main.go

```diff
@@ -58,6 +59,15 @@ func main() {
 	app.Email = "https://github.com/docker/machine"
 	app.Before = func(c *cli.Context) error {
 		os.Setenv("MACHINE_STORAGE_PATH", c.GlobalString("storage-path"))
+		switch c.GlobalString("ssh-client-type") {
+		case "gonative":
```

Comment:
Hm, I think it would be difficult for me to divine what either of those means just by looking at them.  I tried to name them to be very explicit about what they are.  @ehazlett WDYT?